packages/hoppscotch-app/components/smart/Tabs.vue

```diff
@@ -113,53 +128,73 @@ const emit = defineEmits<{
 
-const addTabEntry = (tabID: string, meta: TabMeta) => {
-  tabEntries.value = pipe(
-    tabEntries.value,
-    O.fromPredicate(not(A.exists(([id]) => id === tabID))),
-    O.map(A.append([tabID, meta] as [string, TabMeta])),
-    O.getOrElseW(() => throwError(`Tab with duplicate ID created: '${tabID}'`))
-  )
+provide<TabProvider>("tabs-system", {
+  renderInactive: computed(() => props.renderInactiveTabs),
+  activeTabID: computed(() => props.value),
+})
+
+const selectTab = (id: string) => {
+  emit("input", id)
 }
 
-const updateTabEntry = (tabID: string, newMeta: TabMeta) => {
-  tabEntries.value = pipe(
-    tabEntries.value,
-    A.findIndex(([id]) => id === tabID),
-    O.chain((index) =>
-      pipe(
-        tabEntries.value,
-        A.updateAt(index, [tabID, newMeta] as [string, TabMeta])
+type AnyComponent = VNode["componentInstance"]
+type TabComponent = InstanceType<typeof Tab>
+
+const isTabComponent = (
+  componentInstance: AnyComponent | TabComponent
+): componentInstance is TabComponent =>
+  !!(componentInstance && "implementsTab" in componentInstance)
+
+const getComponentFromNode = (node: VNode) =>
+  node.componentInstance as AnyComponent | TabComponent
+
+/**
+ * proxy is the only allowed key to be used from getCurrentInstance
+ * see https://github.com/vuejs/vue/issues/12596#issuecomment-1173269807
+ */
+const instance = getCurrentInstance()?.proxy
+
+const getTabs = () => {
+  const nodes = instance?.$slots?.default
+
+  const tabs = nodes
+    ? pipe(
+        nodes,
+        A.map(getComponentFromNode),
+        A.filter(isTabComponent),
+        A.map(({ $props }): [string, TabMeta] => [
+          $props.id,
+          {
+            icon: $props.icon,
+            label: $props.label,
+            indicator: $props.indicator,
+            info: $props.info,
+          },
+        ])
       )
-    ),
-    O.getOrElseW(() => throwError(`Failed to update tab entry: ${tabID}`))
-  )
+    : []
+
+  if (!isEqual(tabEntries.value, tabs)) {
+    tabEntries.value = tabs
+  }
 }
 
-const removeTabEntry = (tabID: string) => {
-  tabEntries.value = pipe(
-    tabEntries.value,
-    A.findIndex(([id]) => id === tabID),
-    O.chain((index) => pipe(tabEntries.value, A.deleteAt(index))),
-    O.getOrElseW(() => throwError(`Failed to remove tab entry: ${tabID}`))
-  )
-
-  // If we tried to remove the active tabEntries, switch to first tab entry
-  if (props.value === tabID)
-    if (tabEntries.value.length > 0) selectTab(tabEntries.value[0][0])
+const setActiveTab = () => {
+  const hasActiveTab = tabEntries.value.some((tab) => tab[0] === props.value)
+
+  if (!hasActiveTab && tabEntries.value.length > 0) {
+    selectTab(tabEntries.value[0][0])
+  }
 }
 
-provide<TabProvider>("tabs-system", {
-  renderInactive: computed(() => props.renderInactiveTabs),
-  activeTabID: computed(() => props.value),
-  addTabEntry,
-  updateTabEntry,
-  removeTabEntry,
+onUpdated(() => {
+  getTabs()
+  setActiveTab()
 })
```

Comment:
`onUpdated` gets called on updates on children as well it seems as well as when tab is switched